Riot/Modules/MatrixKit/Models/Room/MXKRoomDataSource.m

```diff
@@ -2185,10 +2185,32 @@ - (void)resendEventWithEventId:(NSString *)eventId success:(void (^)(NSString *)
             [self removeEventWithEventId:eventId];
             
             if (event.isVoiceMessage) {
-                NSNumber *duration = event.content[kMXMessageContentKeyExtensibleAudioMSC1767][kMXMessageContentKeyExtensibleAudioDuration];
-                NSArray<NSNumber *> *samples = event.content[kMXMessageContentKeyExtensibleAudioMSC1767][kMXMessageContentKeyExtensibleAudioWaveform];
-                
-                [self sendVoiceMessage:localFileURL mimeType:mimetype duration:duration.doubleValue samples:samples success:success failure:failure];
+                // Check if it is an actual voice message or a voicebroadcast chunk
+                if (event.content[VoiceBroadcastSettings.voiceBroadcastContentKeyChunkType] != nil) {
+                    // VoiceBroadcast chunk
+                    NSNumber *duration = event.content[kMXMessageContentKeyExtensibleAudioMSC1767][kMXMessageContentKeyExtensibleAudioDuration];
+                    NSDictionary* additionalContentParams = @{
+                        kMXEventRelationRelatesToKey: event.content[kMXEventRelationRelatesToKey],
+                        VoiceBroadcastSettings.voiceBroadcastContentKeyChunkType: event.content[VoiceBroadcastSettings.voiceBroadcastContentKeyChunkType]
```

Comment:
You should probably check these values for nil, otherwise you will get crashes when trying to add them to the dictionary